# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/storage/SessionStorageManager.h
# Commit: c11ce752a12e
# Full Hash: c11ce752a12ef6db0e2422858fe2b00471a9ed9b
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-11 03:34:51
# Regressor Bug: 1593246
# File Overlap Count: 1
# Description:
#   Bug 1593246 - Part 2: Give SessionStorageManager a reference to BrowsingContext r=sg
#   
#   This reference is necessary when sending session storage data for
#   all browsing context to the parent process. Note that it entails
#   making SessionStorageManager a cycle collection participant, since
# ==============================================================================

diff -r 433fd567ff92 -r c11ce752a12e dom/storage/SessionStorageManager.h
--- a/dom/storage/SessionStorageManager.h	Tue Dec 10 12:12:25 2019 +0000
+++ b/dom/storage/SessionStorageManager.h	Tue Dec 10 12:12:32 2019 +0000
@@ -21,12 +21,18 @@
 class SessionStorageManager final : public nsIDOMSessionStorageManager,
                                     public StorageObserverSink {
  public:
-  SessionStorageManager();
+  explicit SessionStorageManager(RefPtr<BrowsingContext> aBrowsingContext);
 
-  NS_DECL_ISUPPORTS
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_NSIDOMSTORAGEMANAGER
   NS_DECL_NSIDOMSESSIONSTORAGEMANAGER
 
+  NS_DECL_CYCLE_COLLECTION_CLASS(SessionStorageManager)
+
+  RefPtr<BrowsingContext> GetBrowsingContext() const {
+    return mBrowsingContext;
+  }
+
  private:
   ~SessionStorageManager();
 
@@ -55,6 +61,8 @@
   nsClassHashtable<nsCStringHashKey, OriginKeyHashTable> mOATable;
 
   RefPtr<SessionStorageObserver> mObserver;
+
+  RefPtr<BrowsingContext> mBrowsingContext;
 };
 
 }  // namespace dom